web/styles/left_sidebar.css

```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
     }
 }
 
-.narrows_panel {
-    & li a {
-        margin-top: 1px;
+#views-label-container {
+    cursor: pointer;
+    /* Padding of #private_messages_section_header. */
+    padding: 0 10px 1px 6px;
+    /* The margin-left of .private_messages_container. */
+    margin-left: 6px;
+    /* Space between #views-label-container and #left-sidebar-navigation-list in collapsed state. */
+    margin-right: 10px;
+    margin-top: 3px;
+    outline: none;
+}
 
-        &:hover {
+#left-sidebar-navigation-area {
+    & li.top_left_row {
+        position: relative;
+        padding: 2px 10px 1px $far_left_gutter_size;
+
+        & a:hover {
             text-decoration: none;
         }
     }
+
+    &.collapsed {
+        display: flex;
+        align-items: center;
+        /* 2px space between views and direct messages sections when views section is collapsed. */
+        margin-bottom: 2px;
+
+        /* Only show first 5 views in collapsed state. */
+        .top_left_row:nth-child(n + 6) {
```

Comment:
I'm not a fan of this approach, because it leaves the visible views up to source order in the HTML. I think it would be better to just create a new class for the views that aren't shown in the collapsed state (`hide-in-collapsed-state`), and then with any lucky, that additional class will mean you don't need to resort to using `!important` here.